dev/fattest.simplicity/src/componenttest/rules/repeater/FeatureReplacementAction.java

```diff
@@ -495,9 +497,70 @@ public FeatureReplacementAction withBeta() {
         return this;
     }
 
+    @Override
+    public RepeatTestAction checkIfFeaturesPresent(boolean checkIfFeaturesPresent) {
+        this.checkFeaturesPresent = checkIfFeaturesPresent;
+        return this;
+    }
+
     @Override
     public boolean isEnabled() {
-        return checkEnabled();
+        boolean enabled = checkEnabled();
+        if (enabled && checkFeaturesPresent) {
+            enabled = checkAllFeatureFilesPresent();
+        }
+        return enabled;
+    }
+
+    public boolean checkAllFeatureFilesPresent() {
+        Bootstrap b;
+        try {
+            b = Bootstrap.getInstance();
+        } catch (Exception e) {
+            Log.info(c, "checkAllFeatureFilesPresent", "Skipping action '" + toString() + "' because Bootstrap.getInstance() failed.");
+            return false;
+        }
+        String installRoot = b.getValue("libertyInstallPath");
+
+        Set<String> featuresToCheck = new HashSet<String>();
+        featuresToCheck.addAll(addFeatures);
+        featuresToCheck.addAll(alwaysAddFeatures);
+
+        final String path = installRoot + "/lib/features/";
+        final String oldPrefix = "com.ibm.websphere.appserver.";
+        final String newPrefix = "io.openliberty.";
+        final String newPrefixWithJakarta = "io.openliberty.jakarta.";
+        final String suffix = ".mf";
+
+        //We occasionally create features that only exist for fat tests and will not always be present in the liberty image.
+        // To account for this we'll tolerate a 20% failure ratio.
```

Comment:
Why is this necessary?

I don't think a 20% failure ratio is the right way to do this.

Reply:
If a FAT test has a user feature we'll check if the features in the action are present before the FAT installs them, causing it to skip the repeat; I observed that locally. I put a threshold because I knew keeping a master list of fat user features would be impractical.

I'm happy to use an alternative if you have a suggestion?

Reply:
Resolved because in the new design, we're only doing the check on certain tests which don't have custom test features.